src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -361,9 +368,32 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
   }
 
   function updateHostComponent(current, workInProgress) {
+    let nextProps = workInProgress.pendingProps;
+
+    if (nextProps === null) {
+      invariant(
+        workInProgress.stateNode !== null,
+        'We must have new props for new mounts. This error is likely ' +
+          'caused by a bug in React. Please file an issue.',
+      );
+    } else if (current === null) {
```

Comment:
If `newProps === null && current === null` this would avoid creating a new instance. This is a good optimization in the resume scenario. However if `appendInitialChild` has already been called, it would be called **again** when the resumed node completes.

I'm not sure if it is actually possible to get into that scenario now. I'm try to produce a test case.